packages/cli/src/commands/pipelines/diff.ts

```diff
@@ -126,20 +133,27 @@ export default class PipelinesDiff extends Command {
     const {flags} = await this.parse(PipelinesDiff)
     const targetAppName = flags.app
 
-    const coupling = await getCoupling(this.heroku, targetAppName)
-      .then(res => res.body)
-      .catch(() => {})
-
-    if (!coupling) {
+    let coupling: PipelineCoupling | undefined
+    try {
+      ({body: coupling} = await getCoupling(this.heroku, targetAppName))
+    } catch {
       ux.error(`This app (${targetAppName}) does not seem to be a part of any pipeline`)
       return
     }
 
-    const targetAppId = coupling.app.id!
-    const generation = coupling.generation
+    let pipeline: Pipeline
+    try {
+      ({body: pipeline} = await getPipeline(this.heroku, coupling.pipeline!.id!))
+    } catch {
+      ux.error(`Unable to fetch pipeline ${coupling.pipeline!.name}`)
+      return
+    }
```

Comment:
Seems like there could be better consistency with the endpoints. How does a developer know when to use a custom error versus letting the error flow though from the API? When developing this fix, I put this error in intentionally because I didn't find the endpoint error particularly helpful. It was something like "Could not find the pipeline" or "pipeline not found" which wasn't helpful.